crates/bevy_utils/src/packed_option.rs

```diff
@@ -0,0 +1,237 @@
+/// Types that have a specific value which represents a `None` state.
+pub trait NoneValue {
+    const NONE_VALUE: Self;
+    fn is_none_value(&self) -> bool;
+}
+
+macro_rules! impl_max_none_value {
```

Comment:
This macro is only called once.

Reply:
Yea... I left it there in case we want to implement this for other integral types (`u32`, `i64`, etc...), however I'm happy to remove if you think that's better :)

Reply:
I don't have strong feelings on this for now.